aspnet/migration/aspnetcoremigration.rst

```diff
@@ -0,0 +1,190 @@
+
+since HttpPlatformModule was replaced with AspNetCoreModule
+the web.config created by the publish-iis tool now configures IIS to use AspNetCoreModule instead of HttpPlatformHandler to forward requests to Kestrel
+
+The code snippet below shows how to configure the new publish-iis tool in project.json file
+
+.. code-block:: JavaScript
+
+"tools": {
+  "Microsoft.AspNetCore.Server.IISIntegration.Tools": {
+    "version": "1.0.0-*",
+    "imports": "portable-net45+wp80+win8+wpa81+dnxcore50"
+  }
+},
+"scripts": {
+  "postpublish": "dotnet publish-iis --publish-folder %publish:OutputPath% --framework %publish:FullTargetFramework%"
+}
+
+Additionally, you must turn on server garbage collection in ``project.json`` or ``app.config`` when on full .NET framework.
+
+Changes in MVC
+--------------
+
+- Set option in ``project.json`` to preserve the compilation context if you are doing views.
+- You no longer need to reference the Tag Helper package, it's now referenced by ``Razor`` by default.
+
+There are changes that simplify controller discovery:
+There is a new ``Controller`` attribute that can be used to mark a class and their descendants as controllers.
+Classes whose name doesn't end in ``Controller`` and derive from a base class that ends in ``Controller`` are no longer considered controllers. In this scenario the ``[Controller]`` attribute must be applied to the ``Controller`` class itself or to the base class.
+
+We now consider a type to be a controller if all of the following rules apply:
+- The type is a public, concrete, non open generic class.
+- [NonController] is not applied to any type of the hierarchy.
+- The type name ends with ``Controller``, or if the ``[Controller]`` attribute is applied to the type or to one of its ancestors.
+- It's important to note that if [NonController] is applied anywhere in the type hierarchy the discovery conventions will never consider that type or its descendants to be a controller. ``[NonController]`` takes precedence over ``[Controller]``.
+
+
+ViewComponents changes
+----------------------
+
+-The Sync APIs have been removed.
+-``InvokeAsync`` takes an anonynmous object instead of params.
+-``web`` is no longer a thing, use dotnetrun or ``dotnet <dll name>`` at the command prompt instead.
+
+Mapping commands to tools
+-------------------------
+
+The following commands are now tools: Secret manager, watch, sqlcache. You can configure these in the ``tools`` section in the ``project.json`` file found in the root of your project. 
+
+Configuration
+-------------
+
+``IConfigurationSource`` has been introduced to represent the settings/configuration which is used to Build() an IConfigurationProvider. It is no longer possible to access the provider instances from IConfigurationBuilder only the sources. This is intentional, but may cause loss of functionality as you can longer do things like explicitly call Load() on the provider instances.
```

Comment:
> Build()

Nit: Interjecting the code is a little confusing when reading past. Perhaps just say "build" and let the dev see what you have below.

> It is no longer possible to access the provider instances from IConfigurationBuilder only the sources.

Nit: That "only" ... mmm ... maybe ...

... IConfigurationBuilder, only from the sources.

Except that I personally favor breaking those thoughts out ...

It is no longer possible to access the provider instances from IConfigurationBuilder. The provider instances may only be accessed from their sources. ??? Something like that.

> This is intentional, but may cause loss of functionality as you can longer do things like explicitly call Load() on the provider instances.

Nit: "This" is probably the subject of both "is ..." and "may cause ...," so perhaps drop the comma. Besides, the first phrase as a sentence is only a scant three words long. "do things like" is a bit informal. The content behind the "as" is an independent clause, which makes "as" a conjunction, so it will need a comma. Should be "no longer," right? How about this ...

This is intentional but may cause a loss of functionality, as you can no longer explicitly call Load() on the provider instances.